From 9f87b579ec7e23b527bf5f5941188cc8c23dc78d Mon Sep 17 00:00:00 2001
From: Marvin Lin <milkfafa@gmail.com>
Date: Fri, 10 May 2024 10:48:53 +0800
Subject: [PATCH 05/11] media: nuvoton: Fix VCD_STAT_BUSY timeout issue when
 ECC is enabled

Increase VCD_TIMEOUT_US to prevent VCD_STAT_BUSY timeout issue when
ECC is enabled.

Signed-off-by: Marvin Lin <milkfafa@gmail.com>
Upstream-Status: Pending [Not submitted to upstream yet]
---
 drivers/media/platform/nuvoton/npcm-regs.h  | 2 +-
 drivers/media/platform/nuvoton/npcm-video.c | 3 +--
 2 files changed, 2 insertions(+), 3 deletions(-)

diff --git a/drivers/media/platform/nuvoton/npcm-regs.h b/drivers/media/platform/nuvoton/npcm-regs.h
index 6fca5f36872b..2c30e7142ec4 100644
--- a/drivers/media/platform/nuvoton/npcm-regs.h
+++ b/drivers/media/platform/nuvoton/npcm-regs.h
@@ -77,7 +77,7 @@
 
 #define VCD_FB_SIZE			0x500000 /* support up to 1920 x 1200 */
 #define VCD_KVM_BW_PCLK			120000000UL
-#define VCD_TIMEOUT_US			300000
+#define VCD_TIMEOUT_US			1000000
 
 /* ECE Registers */
 #define ECE_DDA_CTRL			0x0000
diff --git a/drivers/media/platform/nuvoton/npcm-video.c b/drivers/media/platform/nuvoton/npcm-video.c
index bdd55965b696..b806e18f37b5 100644
--- a/drivers/media/platform/nuvoton/npcm-video.c
+++ b/drivers/media/platform/nuvoton/npcm-video.c
@@ -851,6 +851,7 @@ static int npcm_video_start_frame(struct npcm_video *video)
 		return 0;
 	}
 
+	npcm_video_vcd_state_machine_reset(video);
 	regmap_read(vcd, VCD_STAT, &status);
 	ret = regmap_read_poll_timeout(vcd, VCD_STAT, val, !(val & VCD_STAT_BUSY),
 				       1000, VCD_TIMEOUT_US);
@@ -871,8 +872,6 @@ static int npcm_video_start_frame(struct npcm_video *video)
 	set_bit(VIDEO_CAPTURING, &video->flags);
 	mutex_unlock(&video->buffer_lock);
 
-	npcm_video_vcd_state_machine_reset(video);
-
 	regmap_update_bits(vcd, VCD_INTE, VCD_INTE_DONE_IE | VCD_INTE_IFOT_IE |
 			   VCD_INTE_IFOR_IE, VCD_INTE_DONE_IE | VCD_INTE_IFOT_IE |
 			   VCD_INTE_IFOR_IE);
-- 
2.43.0

